js/apps/system/_admin/aardvark/APP/react/src/views/visgraphs/GraphContext.tsx

```diff
@@ -0,0 +1,266 @@
+  edges: DataSet<EdgeDataType>;
+};
+type GraphContextType = {
+  graphData: VisGraphData | undefined;
+  graphName: string;
+  onApplySettings: (updatedParams?: { [key: string]: string }) => void;
+  onRestoreDefaults: () => void;
+  network?: Network;
+  isSettingsOpen?: boolean;
+  isGraphLoading?: boolean;
+  fetchDuration?: number;
+  datasets?: DatasetsType;
+  setDatasets: (datasets: DatasetsType) => void;
+  rightClickedEntity?: ActionEntityType;
+  setRightClickedEntity: (
+    rightClickedEntity: ActionEntityType | undefined
+  ) => void;
+  selectedAction?: SelectedActionType;
+  setSelectedAction: (selectedAction: SelectedActionType | undefined) => void;
+  setNetwork: (network: Network) => void;
+  toggleSettings: () => void;
+  onCloseSettings: () => void;
+  onClearAction: () => void;
+  onAddEdge: (args: AddEdgeArgs) => void;
+  selectedEntity?: SelectedEntityType;
+  onSelectEntity: (data: SelectedEntityType) => void;
+  hasDrawnOnce: MutableRefObject<boolean>;
+  graphError?: { code: number; response: { body: { errorMessage: string } } };
+};
+
+const GraphContext = createContext<GraphContextType>({
+  graphName: ""
+} as GraphContextType);
+
+export const fetchVisData = async ({
+  graphName,
+  params
+}: {
+  graphName: string;
+  params: UrlParametersType | undefined;
+}) => {
+  if (!params) {
+    return Promise.resolve();
+  }
+  const data = await getRouteForDB(window.frontendConfig.db, "_admin").get(
+    `/aardvark/visgraph/${graphName}`,
+    params
+  );
+  return data.body;
+};
+
+export const fetchUserConfig = async () => {
+  const username = window.App.currentUser || "root";
+
+  const data = await getRouteForDB(window.frontendConfig.db, "_api").get(
+    `/user/${username}/config`
+  );
+  return data.body.result?.visgraphs;
+};
+
+export const putUserConfig = async ({
+  params,
+  fullConfig,
+  graphName
+}: {
+  params: any;
+  fullConfig: any;
+  graphName: string;
+}) => {
+  const username = window.App.currentUser || "root";
+  const paramKey = `${window.frontendConfig.db}_${graphName}`;
+
+  const finalConfig = {
+    ...fullConfig,
+    [paramKey]: {
+      ...params,
+      mode: ""
+    }
+  };
+  const data = await getRouteForDB(
+    window.frontendConfig.db,
+    "_api"
+  ).put(`/user/${username}/config/visgraphs`, { value: finalConfig });
+  return data.body;
+};
+
+const useSetupParams = ({ graphName }: { graphName: string }) => {
+  const [urlParams, setUrlParams] = useState<UrlParametersType>(
+    DEFAULT_URL_PARAMETERS
+  );
+  const [params, setParams] = useState<UrlParametersType>();
+  useEffect(() => {
+    async function fetchData() {
+      const config = await fetchUserConfig();
+      const paramKey = `${window.frontendConfig.db}_${graphName}`;
+      const graphParams = config?.[paramKey] || DEFAULT_URL_PARAMETERS;
+      setUrlParams(graphParams);
+      setParams(graphParams);
+    }
+    fetchData();
+  }, [graphName]);
+  return { urlParams, setUrlParams, params, setParams };
```

Comment:
We can remove core setters from dependencies.

Reply:
removed from the places I could find